tests/grpc/price_listener.go

```diff
@@ -0,0 +1,62 @@
+package grpc
+
+import (
+	"fmt"
+
+	oracletypes "github.com/umee-network/umee/v4/x/oracle/types"
+)
+
+func listenForPrices(
+	umeeClient *UmeeClient,
+	params oracletypes.Params,
+	chainHeight *ChainHeight,
+) (*PriceStore, error) {
+	priceStore := NewPriceStore()
+	// Wait until the beginning of a median period
+	var beginningHeight int64
+	for {
+		beginningHeight = <-chainHeight.HeightChanged
+		if isPeriodFirstBlock(beginningHeight, params.MedianStampPeriod) {
+			break
+		}
+	}
+
+	// Record each historic stamp when the chain should be recording them
+	for i := 0; i < int(params.MedianStampPeriod); i++ {
+		height := <-chainHeight.HeightChanged
+		if isPeriodFirstBlock(height, params.HistoricStampPeriod) {
+			exchangeRates, err := umeeClient.QueryExchangeRates()
+			fmt.Printf("block %d stamp: %+v\n", height, exchangeRates)
+			if err != nil {
+				return nil, err
+			}
+			for _, rate := range exchangeRates {
+				priceStore.addStamp(rate.Denom, rate.Amount)
```

Comment:
why do we overwrite? Do we want to fetch prices only after median stamp period passes? The we should update the if condition, rather than querying and overwrting